src/realm/__tests__/realmReducer-test.js

```diff
@@ -46,11 +50,16 @@ describe('realmReducer', () => {
         crossRealmBots: action.data.cross_realm_bots,
 
         //
-        // InitialDataUpdateDisplaySettings. Deprecated!
+        // InitialDataUserSettings
         //
-        // TODO(#4933): Use modern `user_settings` object for these.
 
-        twentyFourHourTime: action.data.twenty_four_hour_time,
+        twentyFourHourTime:
+          action.data.user_settings?.twenty_four_hour_time
+          // Fall back to InitialDataUpdateDisplaySettings for servers that
+          // don't support the user_settings_object client capability.
+          /* $FlowIgnore[incompatible-cast]: If `user_settings` is absent,
+             this will be present. */
+          ?? (action.data.twenty_four_hour_time: boolean),
```

Comment:
Let's leave this complication out of the test. After all, there's just one test case -- it goes down one path or the other.

Hmm, in fact, trying that informs me that this `action.data` lacks a `user_settings`! Let's update `eg.action.register_complete` so that it stays representative of the behavior of current servers.